simulation/common/helpers.py

```diff
@@ -762,7 +815,45 @@ def plot_latitudes(coordinates):
     simple_plot_graph(coordinates[:, 1], "Latitudes")
 
 
+def map_array_to_targets(input_array, target_array):
+    """
+
+    Will map an array of values to the non-zero elements (a target) of targets_array.
+    The assertion that len(input_array) and # of targets must match.
+
+    Examples:
+        If input array is [9, 6, 12] and target_array is [0, 1, 1, 0, 1], the output
+    would be [0, 9, 6, 0, 12].
+        If input array is [7, 4, 3, 1] and target_array is [0, 1, 0, 1, 1], then the assertion will
+    fail as there are four elements in input_array and three targets in target_array, thus an error
+    will be raised.
+
+    :param input_array: array of values that will be mapped to the boolean array
+    :param target_array: a boolean array of zero and non-zero values
+    :returns: a new array consisting of the elements of input_array, mapped to the targets values of target_array.
+    :rtype: np.ndarray
+
+    """
+
```

Comment:
Great documentation, especially since this is quite an important function and performs a non-obvious task